torch/_functorch/_aot_autograd/autograd_cache.py

```diff
@@ -380,6 +381,52 @@ def _reduce_tensor(self, tensor):
         return (_ident, (metadata,))
 
 
+@contextlib.contextmanager
+def normalize_placeholder_names(gm: torch.fx.GraphModule):
```

Comment:
FWIW idk if this is the best way to implement this function: I had to hack in a few extra changes to each node like changing the target and used_names in the namespace so that we properly reconstruct the dynamo graph each time. 

I wish it was easier than this, but I can't think of an easier way that isn't considerably slower, i.e. copying the entire graph. Tests seem to pass with this approach, though.